### ISSUE TITLE
neutron-api container is restarting after Nokia Integration

### ISSUE THREAD

**Issue by shivendra1296** [nuagenetworks/nuage-ospdirector#134]
Hi Team,

We have done the Nokia Nuage integration and after integration, stack update is successful. But after that neutron-api container is restarting with the error message as follows:

RHOSP Version 13.0.16
+ sudo -E kolla_set_configs
INFO:__main__:Loading config file at /var/lib/kolla/config_files/config.json
INFO:__main__:Validating config file
INFO:__main__:Kolla config strategy set to: COPY_ALWAYS
INFO:__main__:Copying service configuration files
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/default/neutron-server to /etc/default/neutron-server
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/my.cnf.d/tripleo.cnf to /etc/my.cnf.d/tripleo.cnf
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/neutron/plugin.ini to /etc/neutron/plugin.ini
INFO:__main__:Deleting /etc/neutron/plugins/ml2/ml2_conf.ini
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/neutron/plugins/ml2/ml2_conf.ini to /etc/neutron/plugins/ml2/ml2_conf.ini
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/neutron/plugins/ml2/openvswitch_agent.ini to /etc/neutron/plugins/ml2/openvswitch_agent.ini
INFO:__main__:Deleting /etc/neutron/dhcp_agent.ini
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/neutron/dhcp_agent.ini to /etc/neutron/dhcp_agent.ini
INFO:__main__:Deleting /etc/neutron/l3_agent.ini
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/neutron/l3_agent.ini to /etc/neutron/l3_agent.ini
INFO:__main__:Deleting /etc/neutron/metadata_agent.ini
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/neutron/metadata_agent.ini to /etc/neutron/metadata_agent.ini
INFO:__main__:Deleting /etc/neutron/neutron.conf
INFO:__main__:Copying /var/lib/kolla/config_files/src/etc/neutron/neutron.conf to /etc/neutron/neutron.conf
INFO:__main__:Writing out command to execute
INFO:__main__:Setting permission for /var/log/neutron
++ cat /run_command
+ CMD='/usr/bin/neutron-server --config-file /usr/share/neutron/neutron-dist.conf --config-dir /usr/share/neutron/server --config-file /etc/neutron/neutron.conf --config-file /etc/neutron/plugin.ini --config-dir /etc/neutron/conf.d/common --config-dir /etc/neutron/conf.d/neutron-server --log-file=/var/log/neutron/server.log'
+ ARGS=
+ [[ ! -n '' ]]
+ . kolla_extend_start
++ [[ ! -d /var/log/kolla/neutron ]]
++ mkdir -p /var/log/kolla/neutron
+++ stat -c %a /var/log/kolla/neutron
++ [[ 2755 != \7\5\5 ]]
++ chmod 755 /var/log/kolla/neutron
++ . /usr/local/bin/kolla_neutron_extend_start
+++ [[ -n '' ]]
+++ [[ -n '' ]]
+++ [[ -n '' ]]
++++ whoami
+++ [[ neutron == \r\o\o\t ]]
+ echo 'Running command: '\''/usr/bin/neutron-server --config-file /usr/share/neutron/neutron-dist.conf --config-dir /usr/share/neutron/server --config-file /etc/neutron/neutron.conf --config-file /etc/neutron/plugin.ini --config-dir /etc/neutron/conf.d/common --config-dir /etc/neutron/conf.d/neutron-server --log-file=/var/log/neutron/server.log'\'''
Running command: '/usr/bin/neutron-server --config-file /usr/share/neutron/neutron-dist.conf --config-dir /usr/share/neutron/server --config-file /etc/neutron/neutron.conf --config-file /etc/neutron/plugin.ini --config-dir /etc/neutron/conf.d/common --config-dir /etc/neutron/conf.d/neutron-server --log-file=/var/log/neutron/server.log'
+ exec /usr/bin/neutron-server --config-file /usr/share/neutron/neutron-dist.conf --config-dir /usr/share/neutron/server --config-file /etc/neutron/neutron.conf --config-file /etc/neutron/plugin.ini --config-dir /etc/neutron/conf.d/common --config-dir /etc/neutron/conf.d/neutron-server --log-file=/var/log/neutron/server.log
The following mechanism drivers were not found: set(['nuage_sriov', 'nuage_hwvtep'])

Please if you could advise on the issue.

**Comment by sjabasti** [comment]
Hi

From the looks of it you deployed the upstream neutron-server container image instead of the Nuage neutron-server container image. You can check this by running "sudo docker ps -a | grep neutron" on the controller. If it contains the Nuage version at the end of the container image name, it is running Nuage otherwise it is running upstream neutron. 

The following documentation describes how to make sure the Nuage neutron-server image is added to the config of overcloud deploy. The result of this script should be included in the overcloud-deploy command.
https://github.com/nuagenetworks/nuage-ospdirector/tree/OSPD13/Documentation/6.0_HWVTEP#phase-3-2-pull-the-nuage-containers-from-the-red-hat-catalog

Please also respond with which Nuage version you are deploying with.

Kind Regards
Steven

**Comment by shivendra1296** [comment]
Thanks Steven,

Yes we identified that we are running openstack neutron container and not the nuage one.

We are deploying nuage version 6-0-10.

Regards,
Shivendra

**Comment by shivendra1296** [comment]
Hi Steven,

Just to add further to the below. We have managed to resolve the issue and found that this issue was due to the sequence of templates in our deployment file. Somehow I missed the following note from documentation :(

**Note

The /home/stack/templates/overcloud_images.yaml file should take precedence over this file.**

Thanks a lot for pointing me to the problem.

I am closing this issue now.

Best Regards,
Shivendra
